main/src/com/google/refine/commands/project/SetMetadataCommand.java

```diff
@@ -0,0 +1,80 @@
+/*
+
+Copyright 2010, Google Inc.
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are
+met:
+
+    * Redistributions of source code must retain the above copyright
+notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above
+copyright notice, this list of conditions and the following disclaimer
+in the documentation and/or other materials provided with the
+distribution.
+    * Neither the name of Google Inc. nor the names of its
+contributors may be used to endorse or promote products derived from
+this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,           
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY           
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+package com.google.refine.commands.project;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONException;
+
+import com.google.refine.commands.Command;
+import com.google.refine.model.medadata.IMetadata;
+import com.google.refine.model.medadata.MetadataFactory;
+import com.google.refine.model.medadata.MetadataFormat;
+
+public class SetMetadataCommand extends Command {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws ServletException, IOException {
+
+        try {
+            MetadataFormat format = MetadataFormat.valueOf(request.getParameter("format"));
+            String jsonContent = request.getParameter("jsonContent");
+            
+            if (StringUtils.isBlank(jsonContent)) {
+                respond(response, "error", "json cannot be blank");
+                return;
+            }
+            
+            IMetadata metadata = MetadataFactory.buildMetadata(format);
+            InputStream in = IOUtils.toInputStream(jsonContent, "UTF-8");
+            metadata.loadFromStream(in);
+            metadata.validate();
+            
```

Comment:
It's not clear to me how this metadata is saved to the project - it looks like it is just validated and then it is not used - can you explain what is happening here?

Reply:
/**
 * We are noting save other metadata file for now, only the ProjectMetaData is saved.
 * But we keep this command in case we want to save the medadata which requires to be saved for different reasons. 
 * For example we cannot restore the original metadata if don't save it
 *
 */

Reply:
Is this command currently called anywhere? If so, it should be renamed to something like `ValidateMetadataCommand` otherwise it will cause confusion. If not, it should be deleted.

Reply:
As I mentioned above, it can be used for future metadata format

Reply:
well let's remove it for now, because it is not needed and redundant with the command to set the project metadata. I don't see how it could be useful in the future given the new way we store metadata. We should rather have commands to change column metadata (types, constraints) and this should be exposed in the menu for each column.

Reply:
remvoed